tools/issue-labeler/src/IssueLabelerService/Labelers/PromptLabeler.cs

```diff
@@ -0,0 +1,98 @@
+
+
+        public PromptLabeler(ILogger<LabelerFactory> logger, TriageRag ragService, RepositoryConfiguration config)
+        {
+            _logger = logger;
+            _ragService = ragService;
+            _config = config;
+        }
+
+        public async Task<string[]> PredictLabels(IssuePayload issue)
+        {
+            _logger.LogInformation($"Predicting Labels using Prompt Labeler for issue #{issue.IssueNumber}");
+
+            BinaryData structure = BinaryData.FromString("""
+                {
+                  "type": "object",
+                  "properties": {
+                    "Service": { "type": "string" },
+                    "ServiceConfidenceScore": { "type": "string" },
+                    "Type": { "type": "string" },
+                    "TypeConfidenceScore": { "type": "string" }
+                  },
+                  "required": [ "Service", "ServiceConfidenceScore", "Type", "TypeConfidenceScore" ],
+                  "additionalProperties": false
+                }
+                """);
+
+            // Format the prompt with issue title and description
+            string formattedPrompt = AzureSdkIssueLabelerService.FormatTemplate(
+                _config.LabelUserPrompt,
+                new Dictionary<string, string>
+                {
+                    { "Title", issue.Title },
+                    { "Description", issue.Body }
+                },
+                _logger);
+
+            // Call the AI model with the prompt through the RAG service
+            var result = await _ragService.SendMessageQnaAsync(
+                _config.LabelInstructions,
+                formattedPrompt,
+                _config.LabelModelName,
+                structure);
+
+            _logger.LogInformation($"AI model response: {result}");
+
+            // Deserialize the result
+            var output = JsonSerializer.Deserialize<LabelingOutput>(result);
+
+            if (output == null)
+            {
+                throw new Exception($"Failed to parse AI model response for {issue.RepositoryName} using the Prompt Labeler for issue #{issue.IssueNumber}");
+            }
+
+            // Validate Type
+            if (IsConfidenceScoreAboveThreshold(output.TypeConfidenceScore, _config.ScoreThreshold) || IsValueUnknown(output.Type))
+            {
+                throw new Exception($"Invalid AI model Category response for {issue.RepositoryName} using the Prompt Labeler for issue #{issue.IssueNumber}: {result}");
+            }
+
+            // Validate Service
+            if (IsConfidenceScoreAboveThreshold(output.ServiceConfidenceScore, _config.ScoreThreshold) || IsValueUnknown(output.Service))
+            {
+                throw new Exception($"Invalid AI model Service response for {issue.RepositoryName} using the Prompt Labeler for issue #{issue.IssueNumber}");
+            }
+
+            // Return the predicted labels
+            return [output.Service, output.Type];
```

Comment:
We also need to patch the service label:

if (prediction == "Azure Container Service(ACS)") prediction = "ACS";
if (prediction == "Managed Kubernetes Service(AKS)") prediction = "AKS";